base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -1315,6 +1319,42 @@ where
         Ok(tx_id)
     }
 
+    /// Submit a completed transaction to the Transaction Manager
+    pub async fn submit_transaction(
+        &mut self,
+        transaction_broadcast_join_handles: &mut FuturesUnordered<
+            JoinHandle<Result<u64, TransactionServiceProtocolError>>,
+        >,
+        tx_id: TxId,
+        tx: Transaction,
+        fee: MicroTari,
+        amount: MicroTari,
+    ) -> Result<(), TransactionServiceError>
+    {
+        trace!(target: LOG_TARGET, "Submit transaction ({}) to db.", tx_id);
+        self.db
+            .insert_completed_transaction(tx_id, CompletedTransaction {
+                tx_id,
+                source_public_key: self.node_identity.public_key().clone(),
+                destination_public_key: self.node_identity.public_key().clone(),
+                amount,
+                fee,
+                transaction: tx,
+                status: TransactionStatus::Completed,
+                message: "Submitted Transaction".to_string(),
```

Comment:
It would be useful to make the message an argument for future constructed transaction and then call it with "Coin Split" or some kind of descriptive message

Reply:
Good suggestion, I will make the change.

Reply:
Change applied